### DISCUSSION TITLE
Create user record with OTP only

### DISCUSSION THREAD

**Discussion by cem-al** [pocketbase/pocketbase#5978] [Q&A]
I'm working on a signup flow using only OTP auth.

However, with `Identity/Password` disabled, I get the error:

```
ERROR POST /api/collections/users/records
└─ Failed to create record.
   └─ map[data:password: cannot be blank; passwordConfirm: cannot be blank. raw:password: cannot be blank; passwordConfirm: cannot be blank.]
```
 
 Should I just pass a randomly generated password/passwordConfirm to these fields, or is there another way?

**Comment by chnfyi** [comment]
Yes, you have to set a password when you are creating an auth record. It's recommanded to use `$security.randomString(30)` in a js hook, `security.RandomString(30)` in go hook, or `self.crypto.randomUUID()` if you create the user from browser.

**Comment by ganigeorgiev** [comment]
If you want to create the record automatically with the OTP request you could check https://github.com/pocketbase/pocketbase/discussions/5913.
